Firestore/Source/Auth/FSTCredentialsProvider.m

```diff
@@ -74,28 +67,40 @@ - (instancetype)initWithApp:(FIRApp *)app {
   self = [super init];
   if (self) {
     _app = app;
-    _auth = [FIRAuth authWithApp:app];
-    _currentUser = [[FSTUser alloc] initWithUID:self.auth.currentUser.uid];
+    _currentUser = [[FSTUser alloc] initWithUID:[self.app getUID]];
     _userCounter = 0;
 
     // Register for user changes so that we can internally track the current user.
     FSTWeakify(self);
-    _authListenerHandle = [self.auth addAuthStateDidChangeListener:^(FIRAuth *auth, FIRUser *user) {
-      FSTStrongify(self);
-      if (self) {
-        @synchronized(self) {
-          FSTUser *newUser = [[FSTUser alloc] initWithUID:user.uid];
-          if (![newUser isEqual:self.currentUser]) {
-            self.currentUser = newUser;
-            self.userCounter++;
-            FSTVoidUserBlock listenerBlock = self.userChangeListener;
-            if (listenerBlock) {
-              listenerBlock(self.currentUser);
-            }
-          }
-        }
-      }
-    }];
+    _authListenerHandle = [[NSNotificationCenter defaultCenter]
+        addObserverForName:FIRAuthStateDidChangeInternalNotification
+                    object:nil
+                     queue:nil
+                usingBlock:^(NSNotification *notification) {
+                  FSTStrongify(self);
+                  if (self) {
+                    @synchronized(self) {
+                      NSDictionary *userInfo = notification.userInfo;
+
+                      // ensure we're only notifiying for the current app.
+                      FIRApp *notifiedApp = userInfo[FIRAuthStateDidChangeInternalNotificationAppKey];
+                      if (![_app isEqual:notifiedApp]) {
```

Comment:
Is there a reason to use `_app` instead of `self.app`?  I think we generally prefer using the property version and we are below for other properties (e.g. `self.currentUser`) so it'd be more consistent.

Reply:
Done.